WordPress/UITests/Tests/DashboardTests.swift

```diff
@@ -5,21 +5,19 @@ import XCTest
 class DashboardTests: XCTestCase {
     override func setUpWithError() throws {
         setUpTestSuite()
-
-        try LoginFlow.login(
-            siteUrl: WPUITestCredentials.testWPcomSiteAddress,
-            email: WPUITestCredentials.testWPcomUserEmail,
-            password: WPUITestCredentials.testWPcomPassword
-        )
     }
 
     override func tearDownWithError() throws {
         takeScreenshotOfFailedTest()
     }
 
-    // Disabled for investigation/maintenance: https://github.com/wordpress-mobile/WordPress-iOS/pull/21132
     func testFreeToPaidCardNavigation() throws {
-        try MySiteScreen()
+        try LoginFlow.login(
+            siteUrl: WPUITestCredentials.testWPcomSiteAddress,
+            email: WPUITestCredentials.testWPcomUserEmail,
+            password: WPUITestCredentials.testWPcomPassword,
+            selectedSiteTitle: WPUITestCredentials.testWPcomSecondaryFreeSite
+        )
             .scrollToFreeToPaidPlansCard()
             .verifyFreeToPaidPlansCard()
             .tapFreeToPaidPlansCard()
```

Comment:
Have you considered not moving the entire LoginFlow to avoid duplications? One way to do it would be to have a `LoginFlow.login() -> LoginEpilogueScreen {}` called in setUp as usual, and the tests would be responsible for selecting their appropriate site, something like:

```
    func testFreeToPaidCardNavigation() throws {
        try LoginEpilogueScreen()
            .continueWithSelectedSite(title: WPUITestCredentials.testWPcomSecondaryFreeSite)
            .scrollToFreeToPaidPlansCard()
            .verifyFreeToPaidPlansCard()
            .tapFreeToPaidPlansCard()
            .assertScreenIsLoaded()
            .selectDomain()
            .goToPlanSelection()
            .assertScreenIsLoaded()
    }
```

WDYT?

Reply:
i like this! i didn't want to update login for all the other tests, so i created a new login flow, `loginWithoutSelectingSite()` to handle this case and the tests in this test class can start from `LoginEpilogueScreen()`, changes made in 2060d16f6b03aa7df3dd36260b9436a2c6696d2f